src/test/scala/org/broadinstitute/dsde/workbench/sam/service/UserServiceSpec.scala

```diff
@@ -194,6 +194,63 @@ class UserServiceSpec extends AnyFlatSpec with Matchers with TestSupport with Mo
     registrationDAO.isEnabled(defaultUserId, samRequestContext).unsafeRunSync() shouldBe false
   }
 
+  it should "enable a user in LDAP if they accept the latest Terms of Service" in {
+    createNewEnabledUser()
+    updateTosVersionThenEnableUser(userAcceptsNewTos = true)
+
+    // User should be enabled in LDAP
+    registrationDAO.isEnabled(defaultUserId, samRequestContext).unsafeRunSync() shouldBe true
+  }
+
+  it should "not enable a user in LDAP when they don't accept the latest Terms of Service" in {
+    createNewEnabledUser()
+    updateTosVersionThenEnableUser()
+
+    // User should not be enabled in LDAP
+    registrationDAO.isEnabled(defaultUserId, samRequestContext).unsafeRunSync() shouldBe false
+  }
+
+  it should "enable a user in LDAP when TOS is disabled" in {
+    createNewEnabledUser()
+    updateTosVersionThenEnableUser(tosEnabled = false)
+
+    // User should be enabled in LDAP
+    registrationDAO.isEnabled(defaultUserId, samRequestContext).unsafeRunSync() shouldBe true
+  }
+
+  private def createNewEnabledUser(): Unit = {
+    // create a user
+    val newUser = serviceTosEnabled.createUser(defaultUser, acceptTos = true, samRequestContext).futureValue
+    newUser shouldBe UserStatus(UserStatusDetails(defaultUserId, defaultUserEmail), Map("ldap" -> true, "allUsersGroup" -> true, "google" -> true, "tosAccepted" -> true))
+
+    // it should be enabled
+    dirDAO.isEnabled(defaultUserId, samRequestContext).unsafeRunSync() shouldBe true
+    registrationDAO.isEnabled(defaultUserId, samRequestContext).unsafeRunSync() shouldBe true
+  }
+
+  private def updateTosVersionThenEnableUser(tosEnabled: Boolean = true, userAcceptsNewTos: Boolean = false): Unit = {
```

Comment:
I'm wondering if a better name for this might be setUpUserWithoutNewTosAcceptance() or something but maybe I'm missing what this method actually does.

Reply:
I see, I want to keep the "enable user" part because this change is specific to the enable user API call, but I agree "WithoutNewTosAcceptance" is clearer than just saying "updateTosVersion", since the point of the test is that the user hasn't accepted the latest ToS when they are enabled. Updating!

Reply:
I decided it would be better not to update, because I realize this function is also used for cases where the user accepts the new tos, (hence the userAcceptsNewTos parameter), so it would be confusing to name it "withoutNewTosAcceptance" since that's not what happens in all cases-